### DISCUSSION TITLE
How to Register Custom Component for use in Schema

### DISCUSSION THREAD

**Discussion by adrian-burlacu-SA** [vueform/vueform#301] [Questions] [answered]
Hello, I followed the guide here, but my custom component only shows up when I use it in the form template but not from using its type in the schema:
https://vueform.com/docs/creating-elements

Is there a way to debug the registration and form creation based on schema? I suspect there is a step missing in the documentation.

**Comment by adamberecz** [comment, marked as answer]
Have you used the following way to register the element?
https://vueform.com/docs/creating-elements#registering-elements

What's your custom element's name? When used in schema as a `type` it should be transformed to kebab-case.

**Reply by adrian-burlacu-SA** [reply]
I was missing the registration step because it got lost in the other information. Perhaps it would be better to have it near the top. Thank you!

**Reply by adamberecz** [reply]
Noted
